flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -321,6 +321,8 @@ public abstract class StreamTask<OUT, OP extends StreamOperator<OUT>>
 
     private long initializeStateEndTs;
 
+    @Nullable private FileMergingSnapshotManager fileMergingSnapshotManager;
```

Comment:
Why this field is not initialized ?

Reply:
After re-thinking, I refactor the code to leverage `SubtaskCheckpointCoordinator` to notify `FileMergingSnapshotManager` about the checkpoints, so code changes in `StreamTask` are removed.